docs/Account.md

```diff
@@ -88,23 +91,58 @@ While the interface is agnostic of signature validation schemes, this implementa
 
 Note that although the current implementation works only with StarkKeys, support for Ethereum's ECDSA algorithm will be added in the future.
 
-### Signer utility
+### Signer
+
+The signer is responsible for creating a transaction signature with the user's private key for a given transaction. This implementation utilizes [Nile's Signer](https://github.com/OpenZeppelin/nile/blob/main/src/nile/signer.py) class to create transaction signatures through the `Signer` method `sign_transaction`.
+
+`sign_transaction` expects the following parameters per transaction:
+
+* `sender` the contract address invoking the tx
+* `calls` a list containing a sublist of each call to be sent. Each sublist must consist of:
+    1. `to` the address of the target contract of the message
+    2. `selector` the function to be called on the target contract
+    3. `calldata` the parameters for the given `selector`
+* `nonce` an unique identifier of this message to prevent transaction replays. Current implementation requires nonces to be incremental
+* `max_fee` the maximum fee a user will pay
+
+Which returns:
+
+* `calls` a list of calls to be bundled in the transaction
+* `calldata` a list of arguments for each call
+* `sig_r` the transaction signature
+* `sig_s` the transaction signature
+
+While the `Signer` class performs much of the work for a transaction to be sent, it neither manages nonces nor invokes the actual transaction on the Account contract. Those functions can be done manually; however, this implementation abstracts that all away with `ActivatedSigner`.
 
-The `Signer()` class in [utils.py](../tests/utils.py) is used to perform transactions on a given Account, crafting the tx and managing nonces.
+### ActivatedSigner utility
 
-It exposes three functions:
+The `ActivatedSigner` class in [utils.py](../tests/utils.py) is used to perform transactions on a given Account, crafting the tx and managing nonces. In order for a transaction to be sent, this utility performs the following:
 
-* `def sign(message_hash)` receives a hash and returns a signed message of it
-* `def send_transaction(account, to, selector_name, calldata, nonce=None, max_fee=0)` returns a future of a signed transaction, ready to be sent.
-* `def send_transactions(account, calls, nonce=None, max_fee=0)` returns a future of batched signed transactions, ready to be sent.
+* checks nonce
+  * if none is given, it fetches the nonce from the Account contract via `get_nonce`
 
-To use Signer, pass a private key when instantiating the class:
+* reformats callarray
+  * a necessary process to convert the `to` contract address to hexadecimal format
```

Comment:
why is it necessary?

Reply:
Because Nile converts `to` to base-16 which requires the `to` address to be a string. Converting the address with `str()` creates an integer that exceeds Cairo's `FIELD_PRIME`...I'll add this to the docs :)